utils.js

```diff
@@ -0,0 +1,58 @@
+import Chapter from './Chapter.js';
+
+/*
+Validate the buyer's name
+parameters: chapterId - the id of the chapter in db
+            nameRu - the buyer's name in Russian (optional)
+            nameHe - the buyer's name in Hebrew (optional)
+returns: the buyer's name in hebrew and russian
+*/
+const validateChapterBuyer = async (chapterId, newNameRu, newNameHe) => {
+  const chapterDoc = await Chapter.findById(chapterId);
+  // If the buyer's name is provided in the query try updating it in the database
+  if (newNameRu || newNameHe) {
+    return await updateBuyer(chapterDoc, newNameRu, newNameHe);
+  } else {
+    return await getBuyer(chapterDoc);
+  }
+};
+
+/*
+Update buyer in the database
+parameters: chapter - mongoose document of the chapter
+            nameRu - the buyer's name in Russian
+            nameHe - the buyer's name in Hebrew
+returns: updated buyer field from the database or throws an error if the buyer already exists
+*/
+const updateBuyer = async (chapter, newNameRu, newNameHe) => {
+  // If the buyer's name is not provided in the database, update it
+  if (isEmpty(chapter.buyer)) {
+    chapter.buyer = {
+      nameRu: newNameRu,
+      nameHe: newNameHe,
+    };
+    await chapter.save();
+    return chapter.buyer;
+    // If the buyer's name is already provided in the database, throw an error
+  } else {
+    throw new Error('Buyer already exists');
+  }
+};
+
+//If the buyer's name is not provided in the query, get it from the database
+const getBuyer = async (chapter) => {
+  if (isEmpty(chapter.buyer)) {
+    return {
```

Comment:
What is the point in returning empty names?

Reply:
because otherwise they'll be undefined and i will need to address this issue (check if they are defined or not) somewhere else
Not sure though what to send via pdf in case if there's no name provided in query and DB
Now i return empty pdf ( that's what empty names are for )